File: docs-old/notintoc/codesnippet/VisualBasic/walkthrough--calling-into-the-visual-studio-sdk-by-using-automation_2.vb
            If commandName = "Addin.Connect.Addin" Then
                Dim sp As IOleServiceProvider = DirectCast(_applicationObject, IOleServiceProvider)
                Dim SID As Guid = GetType(SVsUIShell).GUID
                Dim IID As Guid = GetType(IVsUIShell).GUID
                Dim output As IntPtr
                sp.QueryService(SID, IID, output)
                Dim uiShell As IVsUIShell = DirectCast(Marshal.GetObjectForIUnknown(output), IVsUIShell)

                Dim clsid As Guid = Guid.Empty
                Dim result As Integer
                uiShell.ShowMessageBox(0, clsid, "MyAddin", String.Format(System.Globalization.CultureInfo.CurrentCulture, "Inside " & Me.ToString()), String.Empty, 0, _
                OLEMSGBUTTON.OLEMSGBUTTON_OK, OLEMSGDEFBUTTON.OLEMSGDEFBUTTON_FIRST, OLEMSGICON.OLEMSGICON_INFO, 0, result)

                handled = True
                Return
            End If

File: docs/misc/codesnippet/CSharp/walkthrough-calling-into-the-visual-studio-sdk-by-using-automation_2.cs
                if (commandName == "Addin.Connect.Addin")
                {
                    IOleServiceProvider sp = (IOleServiceProvider)
                      _applicationObject;
                    Guid SID = typeof(SVsUIShell).GUID;
                    Guid IID = typeof(IVsUIShell).GUID;
                    IntPtr output;
                    sp.QueryService(ref SID, ref IID, out output);
                    IVsUIShell uiShell = (IVsUIShell)Marshal.GetObjectForIUnknown(output);

                    Guid clsid = Guid.Empty;
                    int result;
                    uiShell.ShowMessageBox(
                       0,
                       ref clsid,
                       "MyAddin",
                       string.Format(
                          CultureInfo.CurrentCulture, "Inside " + this.ToString()),
                       string.Empty,
                       0,
                       OLEMSGBUTTON.OLEMSGBUTTON_OK,
                       OLEMSGDEFBUTTON.OLEMSGDEFBUTTON_FIRST,
                       OLEMSGICON.OLEMSGICON_INFO,
                       0,
                       out result);

                    handled = true;
                    return;
                }